File: app/src/main/java/net/mmho/photomap2/PhotoViewAdapter.java
package net.mmho.photomap2;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class PhotoViewAdapter extends FragmentPagerAdapter {

    private PhotoGroup group;

    public PhotoViewAdapter(FragmentManager fm,PhotoGroup group) {
        super(fm);
        this.group = group;
    }

    @Override
    public Fragment getItem(int i) {
        Fragment f = new PhotoViewFragment();
        Bundle b = new Bundle();
        b.putLong(PhotoViewFragment.EXTRA_IMAGE_ID,group.get(i).getPhotoId());
        f.setArguments(b);
        return f;
    }

    @Override
    public int getCount() {
        return group.size();
    }

    public long getItemID(int i){
        return group.get(i).getPhotoId();
    }

}

File: app/src/main/kotlin/net/mmho/photomap2/PhotoViewAdapter.kt
package net.mmho.photomap2

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import android.view.ViewGroup

internal class PhotoViewAdapter(fm: FragmentManager, private val group: PhotoGroup?)
        : FragmentPagerAdapter(fm) {

    override fun destroyItem(container: ViewGroup, position: Int, obj:Any) {
        val manager = (obj as Fragment).requireFragmentManager()
        val transaction = manager.beginTransaction()
        transaction.remove(obj)
        transaction.commit()
    }

    override fun getItem(i: Int): Fragment {
        val f = PhotoViewFragment()
        val b = Bundle()
        b.putLong(PhotoViewFragment.EXTRA_IMAGE_ID, getItemID(i))
        f.arguments = b
        return f
    }

    override fun getCount(): Int {
        return group?.size ?: 0
    }

    fun getItemID(i: Int): Long {
        return group?.get(i)?.photo_id ?: 0
    }

}
